packages/flutter_tools/bin/macos_build_flutter_assets.sh

```diff
@@ -40,6 +39,37 @@ if [[ -n "$TRACK_WIDGET_CREATION" ]]; then
   track_widget_creation_flag="--track-widget-creation"
 fi
 
+# Copy the framework and handle local engine builds.
+framework_name = "FlutterMacOS.framework"
+derived_dir="${SOURCE_ROOT}/../build/macos/Build/Products/flutter_framework"
```

Comment:
At some point we should probably unfork this behavior relative to iOS, and put it in the platform directory rather than the build products (updating the example project accordingly). Definitely doesn't need to be now though; maybe when we merge the two scripts, since that'll be a breaking change across the "API" boundary here too.